scripts/ci/gitlab/pipeline/check.yml

```
@@ -35,6 +35,15 @@ test-dependency-rules:
   script:
     - ./scripts/ci/gitlab/ensure-deps.sh
 
+test-rust-features:
+  stage:                           check
+  extends:
+    - .kubernetes-env
```

```suggestion
    - .docker-env
```

I didn't suggest it earlier because I was too optimistic about k8s cluster for this job. It's better to use one of the beefy machines.

---

I just removed the cargo install and do it by searching `Cargo.toml` files. That should be faster.  
The success case is fast anyway, now lets see how slow the failure case is.

---

Okay, let's keep it. We can change the executor at any time.